File: Python/CombinationSumIII/CombinationSumIII.py
class Solution(object):
	# (46ms)
    def combinationSum3(self, k, n):
        """
        :type k: int
        :type n: int
        :rtype: List[List[int]]
        """
        def dfs(ans, now, nowK, nowN):
            if not (nowK or nowN):
                ans.append(now[:])
                return
            if nowK < 0 or nowN < 0:
                return
            for i in xrange(1 if len(now) == 0 else now[-1] + 1, 10):
                now.append(i)
                dfs(ans, now, nowK - 1, nowN - i)
                now.pop()
        ans = []
        dfs(ans, [], k, n)
        return ans

File: JavaScript/CombinationSumIII/CombinationSumIII.js
/**
 * Created by Phoenix-Z on 2017/4/24.
 */
/**
 * (99ms)
 * @param {number} k
 * @param {number} n
 * @return {number[][]}
 */
var combinationSum3 = function(k, n) {
    var ans = [];
    var dfs = function (ans, now, nowK, nowN) {
        if(nowK == 0 && nowN == 0) {
            ans.push(now.slice());
            return;
        }
        if(nowK < 0 || nowN < 0)
            return;
        for(var i = now.length == 0 ? 1 : now[now.length - 1] + 1; i <= 9; i++) {
            now.push(i);
            dfs(ans, now, nowK - 1, nowN - i);
            now.pop();
        }
    };
    dfs(ans, [], k, n);
    return ans;
};